### PR TITLE
Port MASTG-TEST-0003 (by @guardsquare)

### LINKED ISSUE #2935: MASTG v1->v2 MASTG-TEST-0003: Testing Logs for Sensitive Data (android)
Create a new MASTG v2 test covering for":

- **Title:** Testing Logs for Sensitive Data
- **ID:** MASTG-TEST-0003
- **Link:** https://mas.owasp.org/MASTG/tests/android/MASVS-STORAGE/MASTG-TEST-0003/
- **Platform:** android
- **MASVS v1:** ['MSTG-STORAGE-3']
- **MASVS v2:** ['MASVS-STORAGE-2']

**Follow the [guidelines](https://docs.google.com/document/d/1veyzE4cVTSnIsKB1DOPUSMhjXow_MtJOtgHeo5HVoho/edit?usp=sharing)**

### PR DESCRIPTION
**Pull request by serek8**
This PR closes #2935.

### CODE REVIEW

File: tests-beta/android/MASVS-STORAGE/MASTG-TEST-0x03.md
Comment (cpholguera) [inline tests-beta/android/MASVS-STORAGE/MASTG-TEST-0x03.md:1]: This one looks like https://mas.owasp.org/MASTG/tests-beta/android/MASVS-STORAGE/MASTG-TEST-0203/ 

if anything from the MASTG-TEST-0003 is missing in MASTG-TEST-0203 you can add it there. Or if we're missing a different aspect then it can be a new test.

**Review by cpholguera — CHANGES_REQUESTED** [review]

File: tests-beta/android/MASVS-STORAGE/MASTG-TEST-0x03.md
Comment (cpholguera) [reply tests-beta/android/MASVS-STORAGE/MASTG-TEST-0x03.md:1]: @serek8 could you please double check this? Thanks 😊

Comment (serek8) [reply tests-beta/android/MASVS-STORAGE/MASTG-TEST-0x03.md:1]: @cpholguera I think it's a similar case to External Storage tests. There are 3 tests:
1. [MASTG-TEST-0200](https://mas.owasp.org/MASTG/tests-beta/android/MASVS-STORAGE/MASTG-TEST-0200/) (dynamic test)
2. [MASTG-TEST-0201](https://mas.owasp.org/MASTG/tests-beta/android/MASVS-STORAGE/MASTG-TEST-0201/) (dynamic test)
3. [MASTG-TEST-0202](https://mas.owasp.org/MASTG/tests-beta/android/MASVS-STORAGE/MASTG-TEST-0202/) (static test)

As far as I remember we split them because we didn't want to mix static and dynamic in the past. Each test also has `type: [static/dynamic]` metadata to let the users filter out tests we are interested in. If we merge dynamic and static tests into one file, a user won't be able to filter anymore. On the other hand, `type` is an array so we can add both types to a one test, which makes it a bit blurry to decide how to treat it.

This pull request describes a static test while [MASTG-TEST-0203](https://mas.owasp.org/MASTG/tests-beta/android/MASVS-STORAGE/MASTG-TEST-0203/) describes a dynamic test. 

If you prefer, I can just extend `MASTG-TEST-0203`'s steps with a subheader for static steps from this PR but then I would also need to do the same with mitigations.

To sum up I think we should keep this PR as a separate test and assign it an appropriate `type : [static]`. Then a tester can unambiguously say they tested X tests out of Y tests if they want to cover only static tests.

Comment (cpholguera) [reply tests-beta/android/MASVS-STORAGE/MASTG-TEST-0x03.md:1]: After this comment and the offline discussion: let's keep this a static test and try to keep static and dynamic separated. We'll include this in the guidelines and update existing tests.

**Review by cpholguera — APPROVED** [review]
